src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -379,7 +380,8 @@ class ProtobufPlugin implements Plugin<Project> {
             // 'resources' of the output of 'main', in which the source protos are placed.  This is
             // nicer than the ad-hoc solution that Android has, because it works for any extended
             // configuration, not just 'testCompile'.
-            inputs.files getSourceSets()[sourceSetOrVariantName].compileClasspath
+            inputs.files (getSourceSets()[sourceSetOrVariantName].compileClasspath)
+                    .withPathSensitivity(PathSensitivity.NAME_ONLY)
```

Comment:
I suppose this would also be necessary in `setupExtractProtosTask()`?

Reply:
Added!

Reply:
There's actually a lot more places this could possibly be added in `setupExtractIncludeProtosTask` but I was going for a "one thing at a time" PR.